rotary-encoder/rotary_encoder.go

```diff
@@ -0,0 +1,66 @@
+//go:build tinygo
+
+// Implementation based on:
+// https://gist.github.com/aykevl/3fc1683ed77bb0a9c07559dfe857304a
+
+package rotary_encoder
+
+import (
+	"machine"
+)
+
+var (
+	states = []int8{0, -1, 1, 0, 1, 0, 0, -1, -1, 0, 0, 1, 0, 1, -1, 0}
+)
+
+func New(pinA, pinB machine.Pin) *Device {
+	return &Device{pinA: pinA, pinB: pinB, oldAB: 0b00000011}
+}
+
+type Device struct {
+	pinA machine.Pin
+	pinB machine.Pin
+
+	precision int
+
+	oldAB int
+	value int
+}
+
+type Config struct {
+	Precision int
+}
+
+func (enc *Device) Configure(cfg Config) {
+	enc.pinA.Configure(machine.PinConfig{Mode: machine.PinInputPullup})
+	enc.pinA.SetInterrupt(machine.PinRising|machine.PinFalling, enc.interrupt)
+
+	enc.pinB.Configure(machine.PinConfig{Mode: machine.PinInputPullup})
+	enc.pinB.SetInterrupt(machine.PinRising|machine.PinFalling, enc.interrupt)
```

Comment:
Please use `PinToggle` instead of `PinRising|PinFalling`. This bitwise operation is not allowed on most platforms, only on AVR, K210, and stm32.

(I see it's not supported on rp2040, so feel free to make a PR for that).

Reply:
Added `machine.PinToggle` definition for RP2040 in https://github.com/tinygo-org/tinygo/pull/4104

Reply:
Updated to use `machine.PinToggle` as suggested